pkg/kubectl/deployment.go

```diff
@@ -27,24 +27,42 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
-// DeploymentBasicGeneratorV1 supports stable generation of a deployment
-type DeploymentBasicGeneratorV1 struct {
+// BaseDeploymentGenerator: implement the common functionality of
```

Comment:
Optional: follow go comment style everywhere in this PR 
 
```go 
// BaseDeploymentGenerator implements ...
```

https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences